test/parallel/test-repl-uncaught-exception-async.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+// This verifies that adding an `uncaughtException` listener in an REPL instance
+// does not suppress errors in the whole application. Adding such listener
+// should throw.
+
+require('../common');
+const ArrayStream = require('../common/arraystream');
+const repl = require('repl');
+const assert = require('assert');
+
+let accum = '';
+
+const output = new ArrayStream();
+output.write = (data) => accum += data.replace('\r', '');
+
+const r = repl.start({
+  prompt: '',
+  input: new ArrayStream(),
+  output,
+  terminal: false,
+  useColors: false,
+  global: false
+});
+
+r.write(
+  'process.nextTick(() => {\n' +
+  '  process.on("uncaughtException", () => console.log("Foo"));\n' +
+  '  throw new TypeError("foobar");\n' +
+  '});\n'
+);
+r.write(
+  'setTimeout(() => {\n' +
+  '  throw new RangeError("abc");\n' +
+  '}, 1);console.log()\n'
+);
+r.close();
+
+setTimeout(() => {
+  assert(/ERR_INVALID_REPL_INPUT.*(?!Type)RangeError: abc/s.test(accum));
+  assert.strictEqual(process.listenerCount('uncaughtException'), 0);
+}, 2);
```

Comment:
This test passes when run with an executable compiled from the current `master` branch.

Reply:
Ah, I see. Possibly not an issue. Anything in the `setTimeout()` will be swallowed but with these changes, they are not swallowed. Ideally, the test would fail on current master, of course. But I'm not sure how much trouble that's worth going through?

Reply:
I updated the test so that it'll fail in master. It'll now remove the listeners after checking the length and then run the assertions. Otherwise the errors will be swallowed.